server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,233 @@
+                     byte[] bytes = objectMapper.writeValueAsBytes(sortedCacheInfos);
+                     responseBuilder.contentType(APPLICATION_JSON).entity(bytes);
+                  } catch (JsonProcessingException e) {
+                     responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+                  }
+                  return (RestResponse) responseBuilder.build();
+               })
+                .toCompletionStage()
+      , invocationHelper.getExecutor())
+            .thenCompose(Function.identity());
+   }
+
+   private CompletionStage<RestResponse> addOrEditSchema(RestRequest request, boolean create) {
+
+      String schemaName = checkMandatorySchemaName(request);
+
+      ContentSource contents = request.contents();
+      if (contents == null) throw new NoDataFoundException("Schema data not sent in the request");
+
+      AdvancedCache<Object, Object> cache = invocationHelper.getRestCacheManager()
+            .getCache(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, request);
+
+
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+
+      CompletableFuture<Object> putSchema;
+      if (create) {
+         putSchema = cache.putIfAbsentAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.CREATED);
+            } else {
+               builder.status(HttpResponseStatus.CONFLICT);
+            }
+            return result;
+         });
+      } else {
+         putSchema = cache.replaceAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.NOT_FOUND);
+            } else {
+               builder.status(HttpResponseStatus.OK);
+            }
+            return result;
+         });
+      }
+
+      return putSchema
+            .thenCompose(r -> {
+               if(isOkOrCreated(builder)) {
+                  return cache.getAsync(schemaName + ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX);
+               } else {
+                  return CompletableFuture.completedFuture(null);
+               }
+            })
+            .thenApply(validationError -> {
+               if(isOkOrCreated(builder)) {
+                  ProtoSchema protoSchema = new ProtoSchema();
+                  protoSchema.name = schemaName;
+                  if(validationError != null) {
+                     protoSchema.error = createErrorContent(schemaName, (String) validationError);
+                     builder.status(HttpResponseStatus.BAD_REQUEST);
+                  }
+                  byte[] detailsResponse = new byte[0];
+                  try {
+                     detailsResponse = invocationHelper.getMapper().writeValueAsBytes(protoSchema);
+                  } catch (JsonProcessingException e) {
+                     builder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Do we want to at least log the exception?

Reply:
I change this launching an exception, but should not be happening 
it's uneven across resources now, sometimes we launch a throw new RestResponseException(e); and others just catch

Reply:
I understand that different failures may be propagated. But I think we should at least log an exception instead of just catching it and making it impossible to debug.

Reply:
And sorry the previous code doing this should at least log the error as a trace as well. We should not be swallowing exceptions, except for in very rare cases (especially when we can be reporting an error to the user).

Reply:
there is a global issue concerning logging in the REST resources.
Can we discuss this too with @gustavonalle so we can do some follow up PR on error handling and logging strategy, please?

Reply:
There is an error JSON document to report errors, the query endpoint is using it

Reply:
Some endpoints are not implementing this to handle ```JsonProcessingException``` as it is very unlikely this error would be thrown since the JSON is mapped via Jackson based on always the same class

Reply:
I don't think just adding a single trace message needs another PR... And I have heard many a developer say something is very unlikely to impossible, including myself, and it ends up  happening. So I would rather have at least a trace message when I am completely ignoring a checked Exception.

Reply:
Ok, I agree adding log won't hurt  :smile: